### ISSUE TITLE
LoaderOverlay prevents navigation with back and forward toolbar buttons on web

### ISSUE THREAD

**Issue by gasaichandesu** [rodrigobastosv/loading_overlay#42]
Hi,I have recently discovered a strange bug while using a combination of `auto_route` and `loading_overlay`. The thing is, as soon as loader overlay was shown, browser navigation with back and forward arrows is broken
I have created a sample repository showing this bug: https://github.com/gasaichandesu/web_navigation_test

Steps to reproduce:

- Run application in Chrome, make sure navigation is working using BottomNavigationBar, make sure navigation is working using back and forward buttons
- Press 'Show' button on home screen
- Note that navigation via BottomNavigation bar still works, but navigation with back and forward is already broken
- Press 'Hide' button on home screen
- BottomNavigationBar navigation still works, back and forward still does not

https://user-images.githubusercontent.com/80320221/194065609-f0c24437-ec52-4470-926e-5c627365fc21.mov

**Comment by rodrigobastosv** [comment]
Hi @gasaichandesu, thanks so much for using this package. This package uses [BackButtonInterceptor](https://pub.dev/packages/back_button_interceptor) under the hood. That's probably whats causing this issue.

Can you please confirm to me if you still get this error if you pass false to the disableBackButton property?

Thanks!

**Comment by rodrigobastosv** [comment]
I am not very used to how navigator 2.0 works under the hood. But i will take a look at your sample repository to see if i figured it out. Thanks very much!

**Comment by federicopolesello** [comment]
Same issue for me

**Comment by LOCKEDFILE** [comment]
i tried fork this project.
~~~
  loader_overlay:  
    git:
      url: https://github.com/LOCKEDFILE/loading_overlay.git 
~~~

used  `useBackButtonInterceptor` parameter in `LoaderOverlay`/ `GlobalLoaderOverlay`  widget.

~~~dart
LoaderOverlay(
    useBackButtonInterceptor: false,
    ...
),
~~~

`useBackButtonInterceptor` default value is `!kIsWeb`.

so `BackButtonInterceptor` is disable in flutter web .

**Comment by rodrigobastosv** [comment]
Thank you so much for the contribution @LOCKEDFILE. I merged your PR on the main branch. It's already available on the 2.3.0 version on pub.dev.
